myapp/src/day2/Question8/SlideShow.jsx

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from 'react'
+import './styles.css'
+
+const images = [
+  'slide11.jpg',
+  'slide12.jpg',
+  'slide13.jpg',
+  'slide14.jpg',
+  'slide15.jpg',
+  'slide16.jpg',
+  'slide17.jpg',
+]
+const SlideShow = () => {
+  const [currentIndex, setCurrentIndex] = useState(1)
+  const [speed, setSpeed] = useState(1)
+  const [play, setPlay] = useState(false)
+
+  useEffect(() => {
+    let timer
+    if (play) {
+      timer = setTimeout(() => {
+        if (currentIndex === images.length - 1) {
+          setCurrentIndex(1)
+        } else {
+          setCurrentIndex((prev) => prev + 1)
+        }
+      }, speed * 1000)
+      return () => clearTimeout(timer)
+    }
+  }, [play, currentIndex])
```

Comment:
add dependency for speed also. currently it's not updating when the speed changes